src/Basic/NaturalTransformation.lidr

```diff
@@ -186,30 +187,38 @@ The code above is everything we need to define what a natural transformation is.
 >                      (commutativity natTrans1 a b f))
 >         (sym $ associativity cat2 _ _ _ _ (mapMor fun1 a b f) (component natTrans1 b) (component natTrans2 b))))))
 >
-> composeFunctorNatTrans :
+> naturalTransformationHorizontalComposition :
 >      (cat1, cat2, cat3 : Category)
 >   -> (fun1, fun2 : CFunctor cat1 cat2)
+>   -> (fun3, fun4 : CFunctor cat2 cat3)
 >   -> NaturalTransformation cat1 cat2 fun1 fun2
->   -> (fun3 : CFunctor cat2 cat3)
+>   -> NaturalTransformation cat2 cat3 fun3 fun4
 >   -> NaturalTransformation cat1 cat3
 >      (functorComposition cat1 cat2 cat3 fun1 fun3)
->      (functorComposition cat1 cat2 cat3 fun2 fun3)
-> composeFunctorNatTrans _ _ cat3 fun1 fun2 natTrans fun3 =
+>      (functorComposition cat1 cat2 cat3 fun2 fun4)
+> naturalTransformationHorizontalComposition _ _ cat3 fun1 fun2 fun3 fun4 natTrans12 natTrans34 =
 >   MkNaturalTransformation
->     (\a => mapMor fun3 (mapObj fun1 a) (mapObj fun2 a) (component natTrans a))
+>     (\a => compose cat3 (mapObj fun3 (mapObj fun1 a)) (mapObj fun3 (mapObj fun2 a)) (mapObj fun4 (mapObj fun2 a))
+>              (mapMor fun3 (mapObj fun1 a) (mapObj fun2 a) (component natTrans12 a))
+>              (component natTrans34 (mapObj fun2 a)))
 >     (\a, b, f =>
->       trans (sym $ preserveCompose fun3 _ _ _ (component natTrans a) (mapMor fun2 a b f))
->       (trans (cong (commutativity natTrans a b f))
->              (preserveCompose fun3 _ _ _ (mapMor fun1 a b f) (component natTrans b))))
->
-> composeNatTransFunctor :
->      (cat1, cat2, cat3 : Category)
->   -> (fun1 : CFunctor cat1 cat2)
->   -> (fun2, fun3 : CFunctor cat2 cat3)
->   -> NaturalTransformation cat2 cat3 fun2 fun3
->   -> NaturalTransformation cat1 cat3
->     (functorComposition cat1 cat2 cat3 fun1 fun2)
->     (functorComposition cat1 cat2 cat3 fun1 fun3)
-> composeNatTransFunctor _ _ _ fun1 _ _ natTrans = MkNaturalTransformation
->   (\a => component natTrans (mapObj fun1 a))
->   (\a, b, f => commutativity natTrans _ _ (mapMor fun1 a b f))
+>       let
+>         fun3fun1f = mapMor fun3 (mapObj fun1 a) (mapObj fun1 b) (mapMor fun1 a b f)
+>         fun3fun2f = mapMor fun3 (mapObj fun2 a) (mapObj fun2 b) (mapMor fun2 a b f)
+>         fun4fun2f = mapMor fun4 (mapObj fun2 a) (mapObj fun2 b) (mapMor fun2 a b f)
+>         natTrans12a = component natTrans12 a
+>         natTrans12b = component natTrans12 b
+>         fun3natTrans12a = mapMor fun3 (mapObj fun1 a) (mapObj fun2 a) natTrans12a
+>         fun3natTrans12b = mapMor fun3 (mapObj fun1 b) (mapObj fun2 b) natTrans12b
+>         natTrans34fun2a = component natTrans34 (mapObj fun2 a)
+>         natTrans34fun2b = component natTrans34 (mapObj fun2 b)
+>       in
+>         (((sym (associativity cat3 (mapObj fun3 (mapObj fun1 a)) (mapObj fun3 (mapObj fun2 a)) (mapObj fun4 (mapObj fun2 a)) (mapObj fun4 (mapObj fun2 b)) fun3natTrans12a natTrans34fun2a fun4fun2f) `trans`
```

Comment:
we are trying to fit into 120 chars as line length. Do you think you can split these lines to make them fit?